holoviews/plotting/bokeh/callbacks.py

```diff
@@ -690,7 +691,10 @@ def _process_selection_event(self):
         popup_is_callable = callable(popup)
         if popup_is_callable:
             with set_curdoc(self.plot.document):
-                popup = popup(**stream.contents)
+                if inspect.iscoroutinefunction(popup):
+                    popup = await popup(**stream.contents)
+                else:
+                    popup = await asyncio.to_thread(popup, **stream.contents)
```

Comment:
I'm not convinced we should move any and all callbacks to a thread automatically.